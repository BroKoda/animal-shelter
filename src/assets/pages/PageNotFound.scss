@import "../variables-mixins";

.not-found-container {
  position: relative;
  width: 400px;
  margin: 0 auto;

  .not-found-header {
    height: 180px;
    position: relative;

    .not-found-number {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      font-size: 170px;
      font-weight: 600;
      z-index: -1;
      color: $gray;
    }

    .not-found-title {
      z-index: 1;
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      font-size: 35px;
      width: 100%;
    }
  }

  button {
    width: auto;
  }
}