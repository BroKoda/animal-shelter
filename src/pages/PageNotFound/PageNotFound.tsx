import React from 'react'

const PageNotFound = (): JSX.Element => {
  return (
    <h1>Page does not exist</h1>
  )
}

export default PageNotFound