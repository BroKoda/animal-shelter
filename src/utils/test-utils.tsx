import React, { PropsWithChildren } from 'react'
import { render } from '@testing-library/react'
import type { RenderOptions } from '@testing-library/react'
import type { PreloadedState } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import { setupStore } from '../app/testStore'
import type { TestAppStore, TestRootReducer } from '../app/testStore'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<TestRootReducer>
  store?: TestAppStore
}

export function renderWithProviders(
  ui: JSX.Element,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
): any {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}