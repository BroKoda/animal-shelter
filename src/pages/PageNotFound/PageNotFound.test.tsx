import { renderWithProviders } from '../../utils/test-utils'
import { BrowserRouter } from 'react-router-dom'
import PageNotFound from './PageNotFound'

describe('Page not found test', () => {
  it('Should render page not found page', async () => {
    const { container } = renderWithProviders(
      <BrowserRouter>
        <PageNotFound />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})