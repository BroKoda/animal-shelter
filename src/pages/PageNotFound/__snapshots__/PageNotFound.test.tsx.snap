// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page not found test Should render page not found page 1`] = `
<div>
  <div
    class="container-fluid header-container"
  >
    <div
      class="row h-100"
    >
      <div
        class="col-12 align-self-center"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-4 d-md-none align-self-center"
            >
              <i
                class="fa-solid fa-bars js-navigation-toggle"
              />
            </div>
            <div
              class="col-4 col-md-6 align-self-center text-center text-md-start"
            >
              <a
                href="/"
              >
                <img
                  alt="Mancs logo"
                  class="d-none d-md-block"
                  src="/mancs.png"
                />
                <img
                  alt="Mancs logo"
                  class="d-block d-md-none mx-auto"
                  src="/mancs_logo_only.png"
                />
              </a>
            </div>
            <div
              class="col-4 col-md-6 align-self-center text-end"
            >
              <a
                class="d-none d-md-inline-block"
                href="/regisztracio"
              >
                <button
                  class="button secondary-button me-2"
                >
                  Regisztráció
                </button>
              </a>
              <a
                href="/bejelentkezes"
              >
                <button
                  class="button call-to-action-button login-button"
                >
                  <span
                    class="d-none d-md-inline"
                  >
                    Bejelentkezés
                  </span>
                  <span
                    class="d-inline d-md-none"
                  >
                    <i
                      class="fa-solid fa-user"
                    />
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container body-container mt-3 mb-5"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-4 col-lg-3 navigation-container d-none d-md-block pe-md-0 pe-xl-2"
      >
        <div
          class="container menu-container px-0"
        >
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <a
                href="/"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-house"
                    />
                  </div>
                  <span>
                    Főoldal
                  </span>
                </div>
              </a>
              <a
                href="/lakok"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-paw"
                    />
                  </div>
                  <span>
                    Lakóink
                  </span>
                </div>
              </a>
              <a
                href="/hirek"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-newspaper"
                    />
                  </div>
                  <span>
                    Hírek
                  </span>
                </div>
              </a>
              <a
                href="/csatlakozz"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-hand-holding-heart"
                    />
                  </div>
                  <span>
                    Csatlakozz!
                  </span>
                </div>
              </a>
              <a
                href="/kapcsolat"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-location-dot"
                    />
                  </div>
                  <span>
                    Kapcsolat
                  </span>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-12 col-md-8 col-lg-9"
      >
        <div
          class="container body-container px-0"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-12 col-sm-10 col-md-12"
            >
              <div
                class="not-found-container text-center"
              >
                <div
                  class="not-found-header"
                >
                  <span
                    class="not-found-number"
                  >
                    404
                  </span>
                  <span
                    class="not-found-title"
                  >
                    Az oldal nem létezik!
                  </span>
                </div>
                <div
                  class="not-found-text"
                >
                  <p>
                    Jaj ne! Ez az oldal sajnos nem letezik. Kérlek ellenőrizd az oldal elérési címét! Amennyiben úgy érzed, hogy az oldalnak létezni kell, vedd fel velünk a kapcsolatot!
                  </p>
                  <a
                    href="/"
                  >
                    <button
                      class="button call-to-action-button mt-4"
                    >
                      Vissza a főoldalra
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container-fluid footer-container"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="container py-5"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-5 col-lg-4 mb-4 mb-md-0"
            >
              <h5>
                Mancs állatvédő egyesület
              </h5>
              <p
                class="footer-description"
              >
                Tevékenységeink és céljaink magába foglalják az egyhagyott, kidobott, megtalált kölyök és felnőtt állatok mentését, gondozását, nevelését és legfőképp gazdához jutását. Hisszuk, hogy minden állatnak van megfelelő gazdája, mi segítünk megtalálni!
              </p>
            </div>
            <div
              class="col-6 col-md-4 offset-lg-1 useful-links"
            >
              <h5>
                Hasznos linkek
              </h5>
              <a
                href="/"
              >
                Főoldal
              </a>
              <a
                href="/lakok"
              >
                Lakóink
              </a>
              <a
                href="/hirek"
              >
                Hírek
              </a>
              <a
                href="/csatlakozz"
              >
                Csatlakozz!
              </a>
              <a
                href="/kapcsolat"
              >
                Kapcsolat
              </a>
            </div>
            <div
              class="col-6 col-md-3 social-links-container"
            >
              <h5>
                Kapcsolat
              </h5>
              <p>
                Beogradska 3-5
                <br />
                24420 Magyarkanizsa
                <br />
                Szerbia
              </p>
              <h5
                class="mt-4"
              >
                Kövess minket
              </h5>
              <a
                href="/facebook"
                style="text-decoration: none;"
              >
                <i
                  class="fa-brands fa-facebook"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container-fluid copyright-strip-container"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-12 text-center"
            >
              <p>
                © 2023. Minden jog fenntartva. Mancs Állatvédő Egyesület Magyarkanizsa
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
